nncf/experimental/torch/nas/bootstrapNAS/elasticity/elastic_width.py

```diff
@@ -0,0 +1,684 @@
+    def forward(self, _):
+        return self._active_width
+
+
+class DynamicBatchNormInputOp(DynamicWidthOp, nn.Module):
+    SET_RUNNING_STATISTICS = False
+
+    def forward(self, **bn_params):
+        return [param[:self._active_width] for param in bn_params.values()]
+
+
+class ElasticWidthConv2DOp(ElasticWidthOp, nn.Module):
+    def __init__(self, max_out_channels, node_name, elastic_width_params: Dict[str, Any] = None):
+        super().__init__(max_width=max_out_channels, node_name=node_name)
+
+        if elastic_width_params is None:
+            elastic_width_params = {}
+        min_out_channels = elastic_width_params.get('min_out_channels', 32)
+        max_num_params = elastic_width_params.get('max_num_widths', -1)
+        width_step = elastic_width_params.get('width_step', 32)
+        width_multipliers = elastic_width_params.get('width_multipliers', [])
+
+        self._width_list = self.generate_width_list(self._max_width, min_out_channels, max_num_params, width_step,
+                                                    width_multipliers)
+
+    def set_active_width(self, width: int):
+        if width not in self.width_list and width != self.max_width:
+            raise ValueError(f'Invalid number of output channels to set: {width} in scope={self._node_name}. '
+                             f'Should be a number in {self.width_list}')
+        super().set_active_width(width)
+
+    def forward(self, weight, bias):
+        nncf_logger.debug('Conv2d with active width={} in scope={}'.format(self._active_width, self._node_name))
+        num_out_channels = self._active_width
+        new_bias = None if bias is None else bias[:num_out_channels]
+        new_weights = weight[:num_out_channels, :, :, :]
+        return [new_weights, new_bias]
+
+    @staticmethod
+    def generate_width_list(max_out_channels, min_out_channels, max_num_params, width_step, width_multipliers):
+        width_list = []
+        if max_out_channels <= min_out_channels:
+            width_list.append(max_out_channels)
+        elif not width_multipliers:
+            width = max_out_channels
+            width_list.append(width)
+            width -= width_step
+            while width >= min_out_channels:
+                if max_num_params == len(width_list):
+                    break
+                width_list.append(width)
+                width -= width_step
+        else:
+            width_multipliers.sort(reverse=True)
+            if width_multipliers[0] < 1:
+                width_list.append(max_out_channels)
+            for multiplier in width_multipliers:
+                if max_num_params == len(width_list):
+                    break
+                if 0 >= multiplier > 1:
+                    nncf_logger.warning("Wrong value for multiplier: {}. Skipping ".format(multiplier))
+                    continue
+                w = int(max_out_channels * multiplier)
+                w = w - (w % 8)
```

Comment:
Is 8 an alignment constant?

Reply:
The method generates all possible width for the layers to sample from. 8 is just a pre-defined step size and it coincides with the alignment constant.

Makes sense to explicitly extract this constant and add more explanation to doc-strings. Will do that. Thanks for the comment!

Reply:
@daniil-lyakhov extracted constant, added more explanation in docstrings
please review